src/main/java/com/example/springresttask/controller/FileController.java

```diff
@@ -0,0 +1,78 @@
+package com.example.springresttask.controller;
+
+import com.example.springresttask.model.UploadedFile;
+import com.example.springresttask.service.FileService;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+
+@Controller
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+@RequestMapping("/files")
+public class FileController {
+    FileService fileService;
+
+    @GetMapping
+    public String files(Model model) {
+        List<UploadedFile> fileList = fileService.getFileList();
+        model.addAttribute("files", fileList);
+        return "/files";
+    }
+
+    @GetMapping("/upload")
+    public String upload() {
+        return "/upload";
+    }
+
+    @GetMapping("/download/{id}")
+    public String downloadFile(@PathVariable String id) {
```

Comment:
Just use "Integer" as a path variable type. Spring is smart enough to map parameters to respective java types